ndk/src/hardware_buffer.rs

```diff
@@ -131,7 +131,7 @@ impl HardwareBuffer {
         Self { inner: ptr }
     }
 
-    fn as_ptr(&self) -> *mut ffi::AHardwareBuffer {
+    pub fn as_ptr(&self) -> *mut ffi::AHardwareBuffer {
```

Comment:
@zarik5 Since this is a "special case" as you call it, perhaps add a documentation comment explaining how/why the user can use it.

```suggestion
    /// Returns the underlying [`ffi::AHardwareBuffer`] pointer
    ///
    /// The pointer can be used to import this hardware buffer into a Vulkan memory object using [`VK_ANDROID_external_memory_android_hardware_buffer`].
    ///
    /// [`VK_ANDROID_external_memory_android_hardware_buffer`]: https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VK_ANDROID_external_memory_android_hardware_buffer.html
    pub fn as_ptr(&self) -> *mut ffi::AHardwareBuffer {
```

Reply:
That's great, thanks for the suggestion